event-sourcing/event-source/src/main/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotService.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.justice.services.eventsourcing.source.core.snapshot;
+
+import uk.gov.justice.domain.aggregate.Aggregate;
+import uk.gov.justice.domain.snapshot.AggregateChangeDetectedException;
+import uk.gov.justice.domain.snapshot.AggregateSnapshot;
+import uk.gov.justice.domain.snapshot.ObjectInputStreamStrategy;
+import uk.gov.justice.services.eventsourcing.repository.core.SnapshotRepository;
+import uk.gov.justice.services.eventsourcing.source.core.exception.AggregateCreationException;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.commons.lang.SerializationException;
+import org.slf4j.Logger;
+
+/**
+ * The type Default snapshot service.
+ */
+@ApplicationScoped
+public class DefaultSnapshotService implements SnapshotService {
```

Comment:
Is there anyway to move the generics to the class level so we can just use A (or T) everywhere.